test/DockerSdk.Tests/Cli.cs

```diff
@@ -0,0 +1,137 @@
+        /// </summary>
+        /// <param name="commands">An array where each line is a command to run.</param>
+        /// <returns>An array of lines written to stdout across all commands.</returns>
+        /// <remarks>
+        /// If any command returns a non-zero exit code or writes to stderr, the subsequent commands do not run. <br/>
+        /// Use of stdin is not supported.
+        /// </remarks>
+        /// <exception cref="InvalidOperationException">
+        /// One of the commands either wrote to stdout or returned a non-zero exit code.
+        /// </exception>
+        public static string[] Run(params string[] commands)
+            => Run(false, commands);
+
+        /// <summary>
+        /// Runs a series of commands in a Powershell prompt.
+        /// </summary>
+        /// <param name="ignoreErrors">True to ignore any errors that the commands may raise.</param>
+        /// <param name="commands">An array where each line is a command to run.</param>
+        /// <returns>An array of lines written to stdout across all commands.</returns>
+        /// <remarks>
+        /// If any command returns a non-zero exit code or writes to stderr, the subsequent commands do not run, unless
+        /// <paramref name="ignoreErrors"/> is true. <br/> Use of stdin is not supported.
+        /// </remarks>
+        /// <exception cref="InvalidOperationException">
+        /// One of the commands either wrote to stdout or returned a non-zero exit code, and <paramref
+        /// name="ignoreErrors"/> was false.
+        /// </exception>
+        public static string[] Run(bool ignoreErrors, params string[] commands)
+        {
+            var output = new List<string>();
+            foreach (var command in commands)
+                output.AddRange(Run(command, ignoreErrors));
+            return output.ToArray();
+        }
+
+        /// <summary>
+        /// Runs a command in a Powershell prompt.
+        /// </summary>
+        /// <param name="command">The command to run.</param>
+        /// <returns>An array of lines written to stdout.</returns>
+        /// <exception cref="InvalidOperationException">
+        /// The command either wrote to stdout or returned a non-zero exit code.
+        /// </exception>
+        public static string[] Run(string command)
+            => Run(command, false);
+
+        /// <summary>
+        /// Runs a command in a Powershell prompt.
+        /// </summary>
+        /// <param name="command">The command to run.</param>
+        /// <param name="ignoreErrors">True to ignore any errors that the command may raise.</param>
+        /// <returns>An array of lines written to stdout.</returns>
+        /// <exception cref="InvalidOperationException">
+        /// The command either wrote to stdout or returned a non-zero exit code, and <paramref name="ignoreErrors"/> is
+        /// false.
+        /// </exception>
+        public static string[] Run(string command, bool ignoreErrors)
```

Comment:
This would add as requirement having powershell installed for development.
What on exceptions other than System.ComponentModel.Win32Exception?

Reply:
Yeah. I went with PowerShell because it's available on all major platforms with no fuss, and is already installed on the runner.

Reply:
The only other exception that wouldn't be a bug in this code is platform-not-supported. But I don't expect anyone to run the tests on a Nano server :)